core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -99,3 +101,20 @@ private[spark] class SystemClock extends Clock {
     -1
   }
 }
+
+/**
+ * To return a mocked system clock for testing purposes
```

Comment:
the comment seems stale. Also, the class name does not seem to reflect the functionality - "MockedSystemClock extends ManualClock" seems semantically confusing. How about ManualSystemClock ?